### ISSUE TITLE
Add "Search" to the text selection context menu

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#5695]
### User Story

As a user, I want to be able to search a selected text through the context menu, so I can get to my search results in as few steps as possible.

### Deps
- [x] GV bug https://bugzilla.mozilla.org/show_bug.cgi?id=1581161

### Requirements

-Add "search" to the context menu (@jpfaller to provide exact string)
-Add telemetry probe to measure taps on "search"

### Acceptance Criteria

- [ ] When I select a text on a webpage, I see a "search" option in the context menu (this is called "Google Search" in Fennec (see screen shot) and "Web Search" in Chrome
- [ ] When I search for the selected text, a new tab opens with my search results
- [ ] When I search for the selected text in private mode, a new private tab opens with my search results

@jpfaller please note that this issue is different than adding a "search in Fenix" entry to the 3rd party browser context menu, similar to Focus, Opera, DuckDuckGo

![1028964090](https://user-images.githubusercontent.com/42752644/65935867-98fac580-e3e8-11e9-9e0f-a2240d38116a.png)

**Comment by vesta0** [comment]
@lime124 I propose that we skip UX for this issue and refer to the Fennec UI if needed.

**Comment by lime124** [comment]
totally fine to reuse Fennec IxD but we should revisit the "Google Search" string. UR has been showing some conflation in people's mental models around the browser and the search engine. I think this is one spot we can help by reinforcing the  app they are in vs the search engine. @jpfaller for help with this. thanks!

**Comment by liuche** [comment]
This context menu is in GeckoView (because it's in web content), and we'd be able to use the [BasicSelectionActionDelegate](https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/BasicSelectionActionDelegate.html) to override the actions created.

We might need to do this in AC.

**Comment by liuche** [comment]
Passing the action to SearchFragment to handle the Search action makes this a little trickier.

**Comment by ekager** [comment]
FYI to whomever picks this up. Some changes may be happening to the SelectionActionDelegate https://bugzilla.mozilla.org/show_bug.cgi?id=1581161

**Comment by jpfaller** [comment]
Let's make the strings:

Firefox Search
Firefox Private Search

These options should be available when a user selects text OS wide.

We should monitor complaints to see if users are frustrated by this option being available without a way to turn it off.

**Comment by vesta0** [comment]
@liuche @bifleming I just noticed that it was removed from the prioritized backlog 8 days ago but there is no comment as to why that happened. Would you mind adding a note here?

**Comment by liuche** [comment]
I remember gl pulled this into a sprint because he wanted to learn more AC/GV, but I removed it from the sprint when he started working on WebExtensions, however, we can put both of these bugs back!

@gabrielluong I know you were thinking about this bug and did some initial investigation and ran across some complexities. Could you summarize what the approach or complexities would be?

**Comment by liuche** [comment]
If this is similar to #5696 then maybe they can be done sequentially if context is similar.

**Comment by Baron-Severin** [comment]
Spinning out a small story to integrate this work into Fenix https://github.com/mozilla-mobile/fenix/issues/7896.  Should be trivial.

**Comment by Sparky93** [comment]
The above patch referenced this issue erroneously, my patch was for issue https://github.com/mozilla-mobile/fenix/issues/8269

**Comment by sv-ohorvath** [comment]
> Let's make the strings:
> 
> Firefox Search
> Firefox Private Search

@Baron-Severin 
The actual strings contain the app name: Firefox Nightly Search, Firefox Preview Search, but I suppose this is expected and not a hard-coded string? I'm marking this as verified as fixed on Nightly 2/18 and Beta 4.0.

**Comment by lnicola** [comment]
> UR has been showing some conflation in people's mental models around the browser and the search engine. I think this is one spot we can help by reinforcing the app they are in vs the search engine.

I feel conflicted about this. I guess I see the motivation behind it, but I don't need to be reminded which browser I'm using. On the other hand, I might forget what my default search engine is.
